partcad/src/partcad/exception.py

```
@@ -10,3 +10,10 @@
 
 class NeedsUpdateException(Exception):
     pass
+
+class EmptyShapesError(Exception):
+    """Exception raised when no shapes are found for rendering."""
+
+    def __init__(self, message="No shapes found to render. Please specify valid sketches, parts, or assemblies."):
```

Ruff (0.8.2)

Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)